src/regex/pcre2.cr

```diff
@@ -46,14 +46,18 @@ module Regex::PCRE2
     Regex::Options.each do |option|
       if options.includes?(option)
         flag |= case option
-                when .ignore_case?   then LibPCRE2::CASELESS
-                when .multiline?     then LibPCRE2::DOTALL | LibPCRE2::MULTILINE
-                when .extended?      then LibPCRE2::EXTENDED
-                when .anchored?      then LibPCRE2::ANCHORED
-                when .utf_8?         then LibPCRE2::UTF
-                when .no_utf8_check? then LibPCRE2::NO_UTF_CHECK
-                when .dupnames?      then LibPCRE2::DUPNAMES
-                when .ucp?           then LibPCRE2::UCP
+                when .ignore_case?    then LibPCRE2::CASELESS
+                when .multiline?      then LibPCRE2::DOTALL | LibPCRE2::MULTILINE
+                when .dotall?         then LibPCRE2::DOTALL
+                when .extended?       then LibPCRE2::EXTENDED
+                when .anchored?       then LibPCRE2::ANCHORED
+                when .dollar_endonly? then LibPCRE2::DOLLAR_ENDONLY
+                when .firstline?      then LibPCRE2::FIRSTLINE
+                when .utf_8?          then LibPCRE2::UTF
+                when .no_utf8_check?  then LibPCRE2::NO_UTF_CHECK
+                when .dupnames?       then LibPCRE2::DUPNAMES
+                when .ucp?            then LibPCRE2::UCP
+                when .endanchored?    then LibPCRE2::ENDANCHORED
```

Comment:
`.no_jit?` isn't supported yet

Reply:
That constant still needs to be defined in the lib bindings. It is `0x00002000`, which is equal to `NO_AUTO_CAPTURE`, an option valid only for compilation. We will run into problems if we call `pcre2_options` for both compiling and matching

Reply:
...To be honest, if a breaking change is inevitable, we could use this chance to split the public enum into `Regex::CompileOptions` and `Regex::MatchOptions`.

Reply:
PCRE1 does not use the same numerical value for two different options depending on whether compilation or matching happens, so this problem is exclusive to PCRE2. The exact conflicts are:

* the compile flag `Regex::Options::IGNORE_CASE` is shared by the match flag `LibPCRE2::NOTEMPTY_ATSTART`
* the compile flag `Regex::Options::DOTALL` is shared by the match flag `LibPCRE2::PARTIAL_HARD` (by extension `MULTILINE` has this problem too)
* the compile flag `Regex::Options::DOLLAR_ENDONLY` is shared by the match flag `LibPCRE2::PARTIAL_SOFT`
* the match flag `Regex::Options::NO_JIT` is shared by the compile flag `LibPCRE2::NO_AUTO_CAPTURE`

Also, there is no guarantee that the compile flags which do not correspond to any match flags will remain unused for matching in the future.

For the moment, if one tries to pass an incompatible flag we should map that to 0, to avoid any surprising behavior. That means `#pcre2_compile_options`, a copy of this method, will have:

```crystal
# ...
when .no_jit? then 0
```

whereas `#pcre2_match_options` will have something like:

```crystal
case option
when .anchored?       then LibPCRE2::ANCHORED
when .no_utf8_check?  then LibPCRE2::NO_UTF_CHECK
when .endanchored?    then LibPCRE2::ENDANCHORED
when .no_jit?         then LibPCRE2::NO_JIT
when # all remaining members of `Regex::Options`
  0
else
  raise "unreachable"
end
```

Reply:
> For the moment, if one tries to pass an incompatible flag we should map that to 0, to avoid any surprising behavior.

Hm, why not raise? I don't think silently accepting incompatible flags is a good idea.

Reply:
up to you